icssc/AntAlmanac
# Add schedules with their own names

## Summary
- Users can create additional schedules by clicking the + button next to the current schedule's name.
- Once the + button is clicked, a dialog will pop up and prompt the user for the name of the new schedule.
- Once a schedule is added, the user will be redirected to the new schedule and can add courses and custom events.
- If a user has no schedules, they will just be shown one schedule called `Schedule 1`.

## Test Plan
- Add a schedule and see if its name appears in the list of schedules.
- Add courses to the new schedule and see if the Added Courses pane updates.
- Add custom events to the new schedule and see if the Added Courses pane updates.
- Make sure the new schedule does not affect any other schedules.
- Test various schedule functions (finals calendar, undo last deleted course, download as an ics file, screenshot, copy schedule, clear schedule, viewing maps).
- Test if a user with no schedules starts with only `Schedule 1`.

Demo:

https://user-images.githubusercontent.com/78244965/163896748-213fc040-8ddf-4144-b77e-68c1ac5e823d.mp4

## Issues
- Closes #332 
- Closes #333 

## Next Steps
- Allow user to rename schedule
- Allow user to delete schedule
- Redesign the calendar pane toolbar; instead of the new + button, maybe create a central menu for adding a schedule, deleting the current schedule, adding custom events, and renaming the current schedule.
